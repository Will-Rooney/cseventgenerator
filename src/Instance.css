/* Needs improvement */

/* Run Instance Button */
.Run {
	display:inline-block;
	min-width: 100px;
	width: 120px;
	padding:5px 10px;
	background: #0B610B;
	color: #FFFFFF;
	font-weight: bold;
}

/* Terminate Instance Button */
.Terminate {
	display:inline-block;
	min-width: 100px;
	width: 120px;
	padding:5px 10px;
	background: #610B21;
	color: #FFFFFF;
	font-weight: bold;
}
