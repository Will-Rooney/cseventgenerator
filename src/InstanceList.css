.InstanceList {
  margin: 20px;
  padding: 20px;
}

/* Instance Table Format */
#InstanceTable {
    font-family: "Trebuchet MS", Arial, Helvetica, sans-serif;
    border-collapse: collapse;
    width: 100%;
}
#InstanceTable td, #InstanceTable th {
    border: 1px solid #ddd;
    padding: 8px;
}
#InstanceTable tr:nth-child(even){background-color: #f2f2f2;}
#InstanceTable tr:hover {background-color: #ddd;}
#InstanceTable th {
    padding-top: 12px;
    padding-bottom: 12px;
    text-align: left;
    background-color: #4CAF50;
    color: white;
}