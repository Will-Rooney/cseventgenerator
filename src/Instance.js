import React, { Component } from 'react';
import './Instance.css'
import InstanceList from './InstanceList';

class Instance extends Component {
	constructor(props) {
		super(props);
		this.state = { status: 'Stopped' };
		// Bind *this to the run/terminate function calls
		this.runInstance = this.runInstance.bind(this);
		this.terminateInstance = this.terminateInstance.bind(this);
	}
	render() {
		return (
			// For each instance - add a new entry to our table
			<tr className="Instance">
				<td>{this.props.name}</td>
				<td>{this.state.status}</td>
				<td>
					<button className="Run" onClick={this.runInstance}>Run</button>
					<button className="Terminate" onClick={this.terminateInstance}>Terminate</button>
				</td>
			</tr>
		);
	}
	runInstance() {
		// Change state
		this.setState({ status: 'Running' });
		// Send POST - TODO
		// View Response - TODO
	}
	terminateInstance() {
		// Change state
		this.setState({ status: 'Stopped' });
		// Send DELETE - TODO
		// View Response - TODO
	}
}


export default Instance;
