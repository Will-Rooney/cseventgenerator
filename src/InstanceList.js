import React, { Component } from 'react';
import './InstanceList.css';

import Instance from './Instance';

class InstanceList extends Component {
	constructor(props) {
		super(props);
		// TODO - Load static list of instances and their properites
		this.state = { instances: ['instance 01', 'instance 02', 'instance 03'] };
	}
	render() {
		return (
			<div className="InstanceList">
				<table id="InstanceTable">
		        <tr>
		          <th>Instance</th>
		          <th>Run Status</th>
		          <th>Run/Terminate</th>
		        </tr>
		        {this.renderInstances()}
		      </table>
			</div>
		);
	}
	renderInstances() {
		return this.state.instances.map(name => (
			<Instance
				key={name}
				name={name}
			/>
		));
	}

}

export default InstanceList;