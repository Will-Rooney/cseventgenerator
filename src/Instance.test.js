import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import InstanceList from './InstanceList';
import Instance from './Instance';

describe(InstanceList, () => {
  const component = shallow(
    <InstanceList />
  );

  it('renders and matches our snapshot', () => {
    const component = renderer.create(
      <InstanceList />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  // Additional tests go here
});