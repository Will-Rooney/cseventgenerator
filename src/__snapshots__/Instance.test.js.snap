// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class InstanceList extends _react.Component {
	constructor(props) {
		super(props);
		this.state = { instances: ['instance 01', 'instance 02', 'instance 03'] };
	}
	render() {
		return (
			_react2.default.createElement('div', { className: 'InstanceList', __source: { fileName: _jsxFileName, lineNumber: 14 }, __self: this },
				this.renderInstances()));


	}
	renderInstances() {
		return this.state.instances.map(name =>
		_react2.default.createElement(_Instance2.default, {
			key: name,
			name: name, __source: { fileName: _jsxFileName, lineNumber: 21 }, __self: this }));


	}} renders and matches our snapshot 1`] = `
<div
  className="InstanceList"
>
  <div
    className="Instance"
  >
    <p>
      <button
        className="Run"
        onClick={[Function]}
      >
        Run
      </button>
      <button
        className="Terminate"
        onClick={[Function]}
      >
        Terminate
      </button>
          "
      instance 01
      " - 
      Stopped
    </p>
  </div>
  <div
    className="Instance"
  >
    <p>
      <button
        className="Run"
        onClick={[Function]}
      >
        Run
      </button>
      <button
        className="Terminate"
        onClick={[Function]}
      >
        Terminate
      </button>
          "
      instance 02
      " - 
      Stopped
    </p>
  </div>
  <div
    className="Instance"
  >
    <p>
      <button
        className="Run"
        onClick={[Function]}
      >
        Run
      </button>
      <button
        className="Terminate"
        onClick={[Function]}
      >
        Terminate
      </button>
          "
      instance 03
      " - 
      Stopped
    </p>
  </div>
</div>
`;
